osquery/dispatcher/dispatcher.cpp

```diff
@@ -133,16 +142,34 @@ void Dispatcher::joinServices() {
   auto& self = instance();
   DLOG(INFO) << "Thread: " << std::this_thread::get_id()
              << " requesting a join";
-  WriteLock join_lock(self.join_mutex_);
 
-  for (auto& thread : self.service_threads_) {
-    thread->join();
-    DLOG(INFO) << "Service thread: " << thread.get() << " has joined";
+  while (1) {
```

Comment:
Totally a nitpick, but do you mind adding a small comment above each of these loops indicating the exit condition you are expecting?